src/blocks/packages.rs

```diff
@@ -0,0 +1,374 @@
+    pub ignore_phased_updates: bool,
+    pub aur_command: Option<String>,
+}
+
+#[derive(Deserialize, Debug, Clone, Copy, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum PackageManager {
+    Apt,
+    Pacman,
+    Aur,
+    Dnf,
+}
+
+pub async fn run(config: &Config, api: &CommonApi) -> Result<()> {
+    let mut config: Config = config.clone();
+
+    let format = config.format.with_default(" $icon $total.eng(w:1) ")?;
+    let format_singular = config
+        .format_singular
+        .with_default(" $icon $total.eng(w:1) ")?;
+    let format_up_to_date = config
+        .format_up_to_date
+        .with_default(" $icon $total.eng(w:1) ")?;
+
+    // If user provide package manager in any of the formats then consider that also
+    macro_rules! any_format_contains {
+        ($name:expr) => {
+            format.contains_key($name)
+                || format_singular.contains_key($name)
+                || format_up_to_date.contains_key($name)
+        };
+    }
+
+    let apt = any_format_contains!("apt");
+    let aur = any_format_contains!("aur");
+    let pacman = any_format_contains!("pacman");
+
+    if !config.package_manager.contains(&PackageManager::Apt) && apt {
+        config.package_manager.push(PackageManager::Apt);
+    }
+    if !config.package_manager.contains(&PackageManager::Pacman) && pacman {
+        config.package_manager.push(PackageManager::Pacman);
+    }
+    if !config.package_manager.contains(&PackageManager::Aur) && aur {
+        config.package_manager.push(PackageManager::Aur);
+    }
+    if !config.package_manager.contains(&PackageManager::Dnf) && aur {
+        config.package_manager.push(PackageManager::Dnf);
+    }
+
+    let warning_updates_regex = config
+        .warning_updates_regex
+        .as_deref()
+        .map(Regex::new)
+        .transpose()
+        .error("invalid warning updates regex")?;
+    let critical_updates_regex = config
+        .critical_updates_regex
+        .as_deref()
+        .map(Regex::new)
+        .transpose()
+        .error("invalid critical updates regex")?;
+    let ignore_updates_regex = config
+        .ignore_updates_regex
+        .as_deref()
+        .map(Regex::new)
+        .transpose()
+        .error("invalid ignore updates regex")?;
+
+    let mut package_manager_vec = Vec::new();
+
+    for &package_manager in config.package_manager.iter() {
+        let backend = match package_manager {
+            PackageManager::Apt => {
+                Box::new(Apt::new(config.ignore_phased_updates).await?) as Box<dyn Backend>
+            }
+            PackageManager::Pacman => Box::new(Pacman::new().await?) as Box<dyn Backend>,
+            PackageManager::Aur => {
+                Box::new(Aur::new(config.aur_command.clone().unwrap_or_default()))
+                    as Box<dyn Backend>
+            }
+            PackageManager::Dnf => Box::new(Dnf::new()) as Box<dyn Backend>,
+        };
+
+        package_manager_vec.push(backend);
```

Comment:
I think if you inline the `backend` variable and write something like `.push(match ... { });`, you will not have to write ` as Box<dyn Backend>` everywhere, and it will be more compact.

Reply:
Done